go.mod

```diff
@@ -120,3 +120,7 @@ replace (
 	github.com/prometheus/common => github.com/prometheus/common v0.23.0
 	github.com/prometheus/statsd_exporter => github.com/prometheus/statsd_exporter v0.18.1-0.20201124082027-8b2b4c1a2b49
 )
+
+// Contains a patch for 32-bit architectures.
+// TODO(mapno): Remove in next OTel upgrade
+replace github.com/Shopify/sarama => github.com/Shopify/sarama v1.29.1
```

Comment:
Right, that would work too. I thought a replace was more noticeable to remove in the future. I'm not completely sure if it'll be automatically removed in the next upgrade with `go mod tidy`.

Reply:
I asked @sdboyer about this, and it's probably preferable to _not_ do a replace, since that pins us at v1.29.1 forever if we forget to remove the replace. In the other case, our sarama dependency will be upgraded if OTel updates it. 

(It's not clear if it'll actually be removed from the go.mod though, it might be a silent update. But since it's an indirect dependency `go mod tidy` might remove it when it can. We'll have to see I guess 🤷 )

Reply:
Sounds like `go mod tidy` [should get rid of it when it can](https://golang.org/ref/mod#go-mod-tidy):

> go mod tidy may also add or remove // indirect comments on require directives. An // indirect comment denotes a module that does not provide packages imported by packages in the main module. These requirements will be present if the module that imports packages in the indirect dependency has no go.mod file. They may also be present if the indirect dependency is required at a higher version than is implied by the module graph; this usually happens after running a command like go get -u ./....

Reply:
TIL. That's great, thanks.